### PR TITLE
[2201.2.x] Fix NPE in SymbolFinder and ReferenceFinder in worker-flush expression when worker is missing

### LINKED ISSUE #36334: NPE when there is a flush action without a peer worker name
**Description:**

```bal
public function main() {
    var a = flush;
}
```
<details>
  <summary>Stack trace.(Click to expand)</summary>

```
java.lang.NullPointerException
	at io.ballerina.compiler.api.impl.ReferenceFinder.visit(ReferenceFinder.java:1209)
	at org.wso2.ballerinalang.compiler.tree.expressions.BLangWorkerFlushExpr.accept(BLangWorkerFlushExpr.java:55)
	at io.ballerina.compiler.api.impl.ReferenceFinder.find(ReferenceFinder.java:224)
	at io.ballerina.compiler.api.impl.ReferenceFinder.visit(ReferenceFinder.java:340)
	at org.wso2.ballerinalang.compiler.tree.BLangSimpleVariable.accept(BLangSimpleVariable.java:54)
	at io.ballerina.compiler.api.impl.ReferenceFinder.find(ReferenceFinder.java:224)
	at io.ballerina.compiler.api.impl.ReferenceFinder.visit(ReferenceFinder.java:391)
	at org.wso2.ballerinalang.compiler.tree.statements.BLangSimpleVariableDef.accept(BLangSimpleVariableDef.java:50)
	at io.ballerina.compiler.api.impl.ReferenceFinder.find(ReferenceFinder.java:224)
	at io.ballerina.compiler.api.impl.ReferenceFinder.visit(ReferenceFinder.java:298)
	at org.wso2.ballerinalang.compiler.tree.BLangBlockFunctionBody.accept(BLangBlockFunctionBody.java:60)
	at io.ballerina.compiler.api.impl.ReferenceFinder.find(ReferenceFinder.java:224)
	at io.ballerina.compiler.api.impl.ReferenceFinder.visit(ReferenceFinder.java:283)
	at org.wso2.ballerinalang.compiler.tree.BLangFunction.accept(BLangFunction.java:76)
	at io.ballerina.compiler.api.impl.ReferenceFinder.find(ReferenceFinder.java:224)
	at io.ballerina.compiler.api.impl.ReferenceFinder.lambda$visit$2(ReferenceFinder.java:267)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at io.ballerina.compiler.api.impl.ReferenceFinder.visit(ReferenceFinder.java:267)
	at org.wso2.ballerinalang.compiler.tree.BLangCompilationUnit.accept(BLangCompilationUnit.java:101)
	at io.ballerina.compiler.api.impl.ReferenceFinder.find(ReferenceFinder.java:224)
	at io.ballerina.compiler.api.impl.ReferenceFinder.findReferences(ReferenceFinder.java:215)
	at io.ballerina.compiler.api.impl.BallerinaSemanticModel.getReferences(BallerinaSemanticModel.java:283)
	at io.ballerina.compiler.api.impl.BallerinaSemanticModel.references(BallerinaSemanticModel.java:272)
	at org.ballerinalang.langserver.semantictokens.SemanticTokensVisitor.handleReferences(SemanticTokensVisitor.java:674)
	at org.ballerinalang.langserver.semantictokens.SemanticTokensVisitor.addSemanticToken(SemanticTokensVisitor.java:654)
	at org.ballerinalang.langserver.semantictokens.SemanticTokensVisitor.visit(SemanticTokensVisitor.java:134)
	at io.ballerina.compiler.syntax.tree.FunctionDefinitionNode.accept(FunctionDefinitionNode.java:66)
	at io.ballerina.compiler.syntax.tree.NodeVisitor.visitSyntaxNode(NodeVisitor.java:850)
	at org.ballerinalang.langserver.semantictokens.SemanticTokensVisitor.getSemanticTokens(SemanticTokensVisitor.java:107)
	at org.ballerinalang.langserver.semantictokens.SemanticTokensUtils.getSemanticTokens(SemanticTokensUtils.java:64)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$semanticTokensFull$15(BallerinaTextDocumentService.java:624)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:479)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```
</details>

### PR DESCRIPTION
**Pull request by dulajdilshan**
## Purpose
$title

Fixes #36334

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by mohanvive** [comment]
@dulajdilshan can you please sync the latest changes and update the PR.

**Review by DamithaSenevirathne — DISMISSED** [review]

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ReferenceFinder.java [hunk 1]
```diff
@@ -1200,6 +1200,11 @@ public void visit(BLangErrorVariableDef bLangErrorVariableDef) {
 
     @Override
     public void visit(BLangWorkerFlushExpr workerFlushExpr) {
+        // Ignore incomplete worker-flush expressions
+        // Ex: var a = flush;
+        if (workerFlushExpr.workerSymbol == null) {
```
Comment (pubudu91) [inline compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ReferenceFinder.java:1205]: Isn't this null check redundant? There's already one in https://github.com/ballerina-platform/ballerina-lang/pull/36837/files#diff-abafd09e4c17786eb7b550f8eccbeb794ce71bff4e383bdaa9b7ecf203b169aeR1318

Comment (dulajdilshan) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ReferenceFinder.java:1205]: This becomes `null` for the cases like this
```ballerina
public function main() {
    var a = flush;
}
```

Comment (dulajdilshan) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ReferenceFinder.java:1205]: Where do we have the other `null` check?

Comment (pubudu91) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ReferenceFinder.java:1205]: I meant the null check here: https://github.com/ballerina-platform/ballerina-lang/blob/9355b815333b7b8b874bb7245d0a816259f6df45/compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ReferenceFinder.java#L1313

Comment (mohanvive) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ReferenceFinder.java:1205]: @dulajdilshan your comments on this?

Comment (dulajdilshan) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ReferenceFinder.java:1205]: Here the NPE happened due to `workerFlushExpr.workerIdentifier` bine null. Hence, corrected it

**Review by pubudu91 — APPROVED** [review]

**Review by pubudu91 — APPROVED** [review]

**Pull request merged**